interfaces/cython/cantera/test/test_utils.py

```diff
@@ -0,0 +1,113 @@
+import numpy as np
+
+import cantera as ct
+from . import utilities
+
+from cantera._cantera import _py_to_any_to_py
+
+class TestPyToAnyValue(utilities.CanteraTest):
+
+    def check_conversion(self, value, check_type=None):
+        out, held_type = _py_to_any_to_py(value)
+        self.assertEqual(out, value)
+        if check_type is not None:
+            self.assertEqual(held_type, check_type)
+
+    def check_inexact_conversion(self, value, check_type=None):
+        out, held_type = _py_to_any_to_py(value)
+        if isinstance(value, np.ndarray):
+            self.assertEqual(out, value.tolist())
+        else:
+            self.assertEqual(out, list(value))
+        if check_type is not None:
+            self.assertEqual(held_type, check_type)
+
+    def check_raises(self, value, ee, regex):
+        with self.assertRaisesRegex(ee, regex):
+            _py_to_any_to_py(value)
+
+    def test_none(self):
+        self.check_conversion(None, "void")
+
+    def test_set(self):
+        # Sets are converted to lists
+        self.check_inexact_conversion({"a", "b"}, "vector<string>")
+
+    def test_empty_list(self):
+        self.check_conversion([])
+
+    def test_empty_ndarray(self):
+        self.check_inexact_conversion(np.ndarray((0,)))
+
+    def test_empty_dict(self):
+        self.check_conversion({})
+
+    def test_scalar_string(self):
+        self.check_conversion("spam", "string")
+
+    def test_scalar_int(self):
+        self.check_conversion(3, "long int")
```

Comment:
This should be `long`, right?

Reply:
Unfortunately not, as this comes from the C++ core. @speth consistently used `long int` in the C++ core in `AnyMap.h`, but in some instances, it is not possible to use `long int` in Cython: e.g. the cast `<long int>` is possible, whereas `v.isType[long int]()` throws an error. At this point, I believe it's most consistent to use `<long int>` for the type cast, and leave `long` in the other instances.

So I guess the difference is based on coding style (my personal preference would be to just use `long`).

Reply:
While I've been consistent about writing `long int`, I don't have that strong a preference for it. `long int` / `long` are barely used in Cantera outside of `AnyMap.h`/`AnyMap.cpp`, so it wouldn't be that invasive to just change it if other have a particular preference for `long`.

Reply:
I also don't have a preference one way or the other, except a preference for consistency. Whether we do it here or elsewhere, whatever the consistent choice is should be documented somewhere.

Reply:
I looked over the code and as @speth indicated, there are probably less than 50 occurrences of `long int` in all of cantera. It's widely used in `ext` dependencies, so I believe there is some consistency and would propose to leave as is.

Regarding the shortened, `long` occurrence, it is necessary for
```
cbool isType "is" [T]()
```
where I haven't found a way to make it work for a two-word C++ type. I added a comment to document this exception.